examples/webgpu_tsl_vfx_tornado.html

```diff
@@ -0,0 +1,340 @@
+		<link type="text/css" rel="stylesheet" href="main.css">
+	</head>
+	<body>
+
+		<div id="info">
+			<a href="https://threejs.org" target="_blank" rel="noopener">three.js webgpu</a> - VFX Tornado
+			<br>
+			Based on <a href="https://threejs-journey.com/lessons/galaxy-generator" target="_blank" rel="noopener">Three.js Journey</a> lesson
+		</div>
+
+		<script type="importmap">
+			{
+				"imports": {
+					"three": "../build/three.webgpu.js",
+					"three/tsl": "../build/three.webgpu.js",
+					"three/addons/": "./jsm/"
+				}
+			}
+		</script>
+
+		<script type="module">
+
+			import * as THREE from 'three';
+			import { dot, cos, float, min, timerLocal, atan2, uniform, pass, bloom, PI, PI2, color, positionLocal, sin, texture, tslFn, uv, vec2, vec3, vec4 } from 'three/tsl';
+
+			import { OrbitControls } from 'three/addons/controls/OrbitControls.js';
+			import { GUI } from 'three/addons/libs/lil-gui.module.min.js';
+
+			let camera, scene, renderer, postProcessing, controls;
+
+			init();
+
+			function init() {
+
+				camera = new THREE.PerspectiveCamera( 25, window.innerWidth / window.innerHeight, 0.1, 50 );
+				camera.position.set( 1, 1, 3 );
+
+				scene = new THREE.Scene();
+
+				// textures
+
+				const textureLoader = new THREE.TextureLoader();
+				const perlinTexture = textureLoader.load( './textures/noises/perlin/rgb-256x256.png' );
+				perlinTexture.wrapS = THREE.RepeatWrapping;
+				perlinTexture.wrapT = THREE.RepeatWrapping;
+
+				// TSL functions
+
+				const toRadialUv = tslFn( ( [ uv, multiplier, rotation, offset ] ) => {
+
+					const centeredUv = uv.sub( 0.5 ).toVar();
+					const distanceToCenter = centeredUv.length();
+					const angle = atan2( centeredUv.y, centeredUv.x );
+					const radialUv = vec2( angle.add( PI ).div( PI2 ), distanceToCenter ).toVar();
+					radialUv.mulAssign( multiplier );
+					radialUv.x.addAssign( rotation );
+					radialUv.y.addAssign( offset );
+
+					return radialUv;
+
+				} );
+
+				const toSkewedUv = tslFn( ( [ uv, skew ] ) => {
+
+					return vec2(
+						uv.x.add( uv.y.mul( skew.x ) ),
+						uv.y.add( uv.x.mul( skew.y ) )
+					);
+
+				} );
+
+				const twistedCylinder = tslFn( ( [ position, parabolStrength, parabolOffset, parabolAmplitude, time ] ) => {
+
+					const angle = atan2( position.z, position.x ).toVar();
+					const elevation = position.y;
+
+					// parabol
+					const radius = parabolStrength.mul( position.y.sub( parabolOffset ) ).pow( 2 ).add( parabolAmplitude ).toVar();
+
+					// turbulences
+					radius.addAssign( sin( elevation.sub( time ).mul( 20 ).add( angle.mul( 2 ) ) ).mul( 0.05 ) );
+
+					const twistedPosition = vec3(
+						cos( angle ).mul( radius ),
+						elevation,
+						sin( angle ).mul( radius )
+					);
+
+					return twistedPosition;
+
+				} );
+
+				const luminance = tslFn( ( [ color ] ) => {
+
+					return dot( color, vec3( 0.2126, 0.7152, 0.0722 ) );
+
+				} );
```

Comment:
Import `luminance` from `three/tsl` instead.

Reply:
Such a beautiful example!

Reply:
Yeah, I wouldn't be surprised if we see some game using this effect soon :)